src/components/content/order/create/OrderSubmit.tsx

```diff
@@ -29,13 +36,20 @@ function OrderSubmit(state: OrderSubmitProps): React.JSX.Element {
     const [isEulaAccepted, setIsEulaAccepted] = useState<boolean>(false);
     const [isShowDeploymentResult, setIsShowDeploymentResult] = useState<boolean>(false);
     const uniqueRequestId = useRef(v4());
+    const [optionType, setOptionType] = useState<OperationType | undefined>(undefined);
     const submitDeploymentRequest = useDeployRequestSubmitQuery();
-    const getServiceDetailsByIdQuery = useServiceDetailsPollingQuery(
-        submitDeploymentRequest.data?.serviceId ?? '',
+    const redeployFailedDeploymentQuery = useRedeployFailedDeploymentQuery();
+    const getSubmitLatestServiceOrderStatusQuery = useLatestServiceOrderStatusQuery(
+        submitDeploymentRequest.data?.orderId ?? '',
         submitDeploymentRequest.isSuccess,
-        state.serviceHostingType,
-        [serviceDeploymentState.DEPLOYMENT_SUCCESSFUL, serviceDeploymentState.DEPLOYMENT_FAILED]
+        [taskStatus.SUCCESSFUL, taskStatus.FAILED]
     );
+    const getReDeployLatestServiceOrderStatusQuery = useLatestServiceOrderStatusQuery(
```

Comment:
would it be possible to have just one query for both cases?
Ideally the old query must be automatically replaced with new one as soon as the ID is replaced with  new order ID.